doc/mapping/full-text-search.md

```diff
@@ -15,113 +16,94 @@ public class BlogPost
+| .Where(c => c.SearchVector.Matches("Npgsql"))            | [WHERE c."SearchVector" @@ 'Npgsql'](https://www.postgresql.org/docs/current/static/textsearch-intro.html#TEXTSEARCH-MATCHING)
+| .Select(c => EF.Functions.ToTsQuery(c.SearchQuery).ToNegative()) | [SELECT !! to_tsquery(c."SearchQuery")](https://www.postgresql.org/docs/current/static/textsearch-features.html#TEXTSEARCH-MANIPULATE-TSQUERY)
+| .Select(c => EF.Functions.ToTsVector(c.Name).SetWeight(NpgsqlTsVector.Lexeme.Weight.A)) | [SELECT setweight(to_tsvector(c."Name"), 'A')](https://www.postgresql.org/docs/current/static/textsearch-features.html#TEXTSEARCH-MANIPULATE-TSVECTOR)
 
 ## Setting up and querying a full text search index on an entity
-This guide will help you setup the most common case of having a `tsvector` column on a table that updates itself with a trigger and querying it with EF Core.
 
-### Setting up and maintaining the full text search index
-First define an entity as shown below:
+As [the PostgreSQL documentation](https://www.postgresql.org/docs/current/static/textsearch-tables.html) explains, full-text search requires an index to run efficiently. This section will show two ways to do this, both (currently) requiring raw SQL in your migrations. Read the PostgreSQL docs for more information on the different approaches.
+
+### Method 1: Expression index
+
+The simpler method to use full-text search is to set up an expression index. Let's take the following entity:
 
 ```c#
 public class Product
 {
     public int Id { get; set; }
     public string Name { get; set; }
     public string Description { get; set; }
-    public NpgsqlTsVector SearchVector { get; set; }
 }
 ```
 
-and modify the `OnModelCreating` of your `DbContext` class as follows:
+Create a migration which will contain the index creation SQL (`dotnet ef migrations add ...`). At this point, open the generated migration with your editor and add the following:
+
 ```c#
-protected override void OnModelCreating(ModelBuilder modelBuilder)
+protected override void Up(MigrationBuilder migrationBuilder)
 {
-    modelBuilder.Entity<Product>()
-                .HasIndex(p => p.SearchVector)
-                .ForNpgsqlHasMethod("GIN"); // Index method on the search vector (GIN or GIST)
+    migrationBuilder.Sql(@"CREATE INDEX fts_idx ON ""Product"" USING GIN (to_tsvector('english', ""Name"" || ' ' || ""Description""));");
 }
-```
 
-Then use the EF Core tools (as shown below) or Visual Studio to add a migration to create this table in the database:
+protected override void Down(MigrationBuilder migrationBuilder)
+    migrationBuilder.Sql(@"DROP INDEX fts_idx;");
+}
 ```
-dotnet ef migrations add "CreateProductTable"
+
+This will create a full-text search index on the `Name` and `Description` columns. You can query as follows:
+
+```c#
+var context = new ProductDbContext();
+var npgsql = context.Products
+    .Where(p => p.SearchVector.Matches(EF.Functions.ToTsQuery("Npgsql")))
+    .FirstOrDefault();
 ```
```

Comment:
@roji When using the expression index method, there will be no `SearchVector` property to query, instead the entire index expression must be repeated in C# in the LINQ query, so it would look like:

```c#
var context = new ProductDbContext();
var npgsql = context.Products
    .Where(p => EF.Functions.ToTsVector("english", p.Name + " " + p.Description)
                            .Matches(EF.Functions.ToTsQuery("Npgsql")))
    .FirstOrDefault();
```

Reply:
Of course you're right, that was a bit of hasty copy-pasting - amended the PR.

However, unless I'm wrong we can omit the `english` part, can't we?

Reply:
The repetition of the index expression in the where query has to match the index expression exactly, so if you want to remove it, it should be removed in both places, but in my opinion, it's a bad practice to omit the language as the default on PostgreSQL can be changed. It's just good to be explicit about it.